File: sdk/agid-ruby/lib/agid.rb
# frozen_string_literal: true

module Agid
  BASE32_ALPHABET = "0123456789ABCDEFGHJKMNPQRSTVWXYZ"
  PREFIX_LENGTH = 2
  HASH_LENGTH = 10
  TOTAL_LENGTH = 12

  Result = Struct.new(:id, :lat, :lon, :face, keyword_init: true)
  Bounds = Struct.new(:minLat, :maxLat, :minLon, :maxLon, keyword_init: true)

  def self.encode(lat, lon)
    raise NotImplementedError, "wire this package to the AGID reference implementation"
  end

  def self.decode(id)
    nil
  end

  def self.cellBounds(id)
    raise NotImplementedError, "wire this package to the AGID reference implementation"
  end
end

File: sdk/agid-elixir/lib/agid.ex
defmodule Agid do
  @base32_alphabet "0123456789ABCDEFGHJKMNPQRSTVWXYZ"
  @prefix_length 2
  @hash_length 10
  @total_length 12

  def base32_alphabet, do: @base32_alphabet
  def prefix_length, do: @prefix_length
  def hash_length, do: @hash_length
  def total_length, do: @total_length

  def encode(_lat, _lon) do
    {:error, :not_implemented}
  end

  def decode(_id), do: nil

  def cellBounds(_id), do: {:error, :not_implemented}
end
